tests/unit/errors/test_common_functionality.py

```diff
@@ -599,14 +595,8 @@ def test_non_jsonapi_parsing_uses_root_as_errors_array_by_default(
         {"message": "FooMessage", "errors": []},
     ),
 )
-def test_non_jsonapi_parsing_uses_errors_array_if_present(make_response, error_doc):
-    res = make_response(
-        error_doc,
-        422,
-        data_transform=json.dumps,
-        headers={"Content-Type": "application/json"},
-    )
-    err = GlobusAPIError(res.r)
+def test_non_jsonapi_parsing_uses_errors_array_if_present(error_doc):
+    err = construct_error(error_class=GlobusAPIError, body=error_doc, http_status=422)
```

Comment:
I haven't found a place yet where `error_class` does not duplicate the existing default class value. Would you add testing to affirm the default class is `GlobusAPIError` when not specified?

There's no need to remove all places where `error_class` specifies the same default value, but I would prefer it if the default wasn't explicitly overridden, like:

```suggestion
    err = construct_error(body=error_doc, http_status=422)
```

Reply:
Let me add tests to the meta-tests for `_testing` which cover this. I might remove all of the specifications of the default value here too.

The interface changed while I was working on both parts -- originally, `error_class` was required -- and I didn't do a cleanup pass here.

Reply:
I think I like the results of testing this independently, so I'm going to mark it resolved as well.